spec/controllers/api_blocks_controller_spec.rb

```diff
@@ -0,0 +1,37 @@
+require "rails_helper"
+
+RSpec.describe Api::BlocksController do
+  describe "POST :create" do
+    context "if API token is incorrect" do
+      it "returns a 401 status code" do
+        redirection = Redirection.find_by!(slug: "gabe")
+
+        post :create, params: { redirection_slug: redirection.slug }
+
+        expect(response.status).to eq 401
+        expect(Redirection.find_by(slug: redirection.slug)).not_to be_nil
+        expect(BlockedReferrer.count).to eq 0
+      end
+    end
+
+    it "unlinks the redirection" do
+      redirection = create(:redirection, url: "https://foobar.neocities.org")
+
+      request.headers["Authorization"] = "Bearer #{ApiController::API_TOKEN}"
```

Comment:
How does this `API_TOKEN` work in testing? Does it grab the token from the .sample.env?

Reply:
It grabs it from `.env` (which must be created locally — it's gitignored). When Rails boots up, Dotenv kicks in and loads `.env`, and this spec uses a fully-booted Rails environment. So that constant is set.

Reply:
Oh, duh. I was thinking that `.env` didn't work in test mode or something, which is not true.